### PR TITLE
Ensure that driver memory setting is used.

### PR DESCRIPTION
**Pull request by mariusvniekerk**
When a user sets `spark.driver.memory` this normally only applies to spark programs that are launched in cluster mode.  

Since this is a pretty major gotcha it feels safer to just use this setting if specified by the user for the argument that ultimately becomes the `--driver-memory` command line argument.

### CODE REVIEW

File: spylon/spark/launcher.py [hunk 1]
```diff
@@ -438,6 +441,15 @@ def __getitem__(self, key):
     def _set_environment_variables(self):
         """Initializes the correct environment variables for spark"""
         cmd = []
+
+        # special case for driver memory.
+        # Since this can be set erroneously on the standard spark conf we want to be able to just pick that up if
+        # present, since other the setting is IGNORED when starting up the spark daemon.
```
Comment (parente) [inline spylon/spark/launcher.py]: otherwise

File: spylon/spark/launcher.py [hunk 2]
```diff
@@ -306,6 +306,9 @@ def _repr_pretty_(self, p, cycle):
     def __setitem__(self, key, value):
         self._conf_dict[key] = value
 
+    def __getitem__(self, key):
+        return self._conf_dict[key]
```
Comment (parente) [inline spylon/spark/launcher.py]: Is this change related to the others?

Comment (parente) [inline spylon/spark/launcher.py]: NM. _spark_launcher_args.get().

File: tests/test_spark_launcher.py [hunk 3]
```diff
@@ -28,3 +28,10 @@ def test_spark_launcher_multiple_argument():
     c.archives = archive
     c._set_environment_variables()
     assert ('--archives ' + ','.join(archive)) in os.environ['PYSPARK_SUBMIT_ARGS']
+
+
+def test_spark_driver_memory():
+    c = sparklauncher.SparkConfiguration()
+    c.conf.spark.driver.memory = "5g"
+    c._set_environment_variables()
+    assert '--driver-memory 5g' in os.environ['PYSPARK_SUBMIT_ARGS']
```
Comment (parente) [inline tests/test_spark_launcher.py]: Is it worth adding tests for the other special params?

Comment (mariusvniekerk) [inline tests/test_spark_launcher.py]: They all do the same thing.  Dunno
